python/test/unit/language/test_tlx.py

```diff
@@ -62,3 +61,45 @@ def test_add2(BLOCK_SIZE, device):
     ref_out1, ref_out2 = dual_add(x, y, a, b)
     torch.testing.assert_close(output1, ref_out1, check_dtype=False)
     torch.testing.assert_close(output2, ref_out2, check_dtype=False)
+
+
+@pytest.mark.skipif(
+    not is_cuda() or torch.cuda.get_device_capability()[0] < 9,
+    reason="Requires compute capability >= 9 for NV",
+)
+@pytest.mark.parametrize("BLOCK_SIZE", [(1024)])
+def test_alloc_barriers(BLOCK_SIZE, device):
+
+    @triton.jit
+    def add_with_mbarrier(
+        x_ptr,
+        y_ptr,
+        z_ptr,
+        n_elements,
+        BLOCK_SIZE: tl.constexpr,
+    ):
+        pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+
+        bars = tlx.alloc_barriers(10, 2)
+
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        mask = offsets < n_elements
+        x = tl.load(x_ptr + offsets, mask=mask)
+        y = tl.load(y_ptr + offsets, mask=mask)
+        output = x + y
+        tl.store(z_ptr + offsets, output, mask=mask)
+
+    torch.manual_seed(0)
+    size = 98432
+    x = torch.rand(size, device=device)
+    y = torch.rand(size, device=device)
+
```

Comment:
nit: perhaps no need to run the load/store code, as they are not testing the kernel. Just checking ttgir should be good.